### ISSUE TITLE
🐛 If a file with a nested biome config is open when VSCode starts biome reverts to defaults

### ISSUE THREAD

**Issue by mshick** [biomejs/biome#6592]
### VS Code version

1.98.2

### Extension version

3.0.3

### Biome version

2.0.5

### Operating system

- [ ] Windows
- [x] macOS
- [ ] Linux

### Description

It seems that the biome instance this plugin uses ends up getting loaded with no configuration at all if a biome-formattable file is open in the editor when VSCode is started. This only happens if that file happens to be in a folder that uses a nested `biome.json` config file. Folders without nested configs do not trigger the bug.

https://github.com/user-attachments/assets/a81ec89a-0323-4dc4-b143-ce61818d7728

You can also trigger the bug by opening the biome-enabled file quickly after VSCode has started, but before you see output from the biome plugin task.

### Steps to reproduce

1. Create a project with a root config and a nested config. (see repro repo)
2. Open a `*.ts` file.
3. Linting and formatting should load and work normally.
4. Quit VSCode.
5. Reopen VSCode.
6. Notice that your biome linting and formatting has now reverted to a default biome config, double quotes, tabs, no additional rules...

### Expected behavior

I would expect the plugin and biome to behave the same regardless of what files are open in my editor when it starts.

### Does this issue occur when using the CLI directly?

No

### Link to a minimal reproduction

https://github.com/mshick/biome-repro-1750955832360/tree/main

### Logs

```plain text

```

**Comment by rikurauhala** [comment]
I also have this issue on Ubuntu 24.04.

**Comment by nhedger** [comment]
Configuration resolution is done at the LSP level, moving the issue to the main repo.
